include/pf-applications/grain_tracker/tracking.h

```diff
@@ -64,44 +62,46 @@ namespace GrainTracker
       {
         ContainerType boxes;
 
-        std::vector<unsigned int> new_segments_to_grains;
+        std::vector<unsigned int> old_segments_to_grains;
 
         for (const unsigned int grain_id : new_grains_per_op[op])
-          for (const auto &new_segment : new_grains.at(grain_id).get_segments())
-            {
-              new_grains_to_old.try_emplace(
-                grain_id, std::numeric_limits<unsigned int>::max());
+          new_grains_to_old.try_emplace(
+            grain_id, std::numeric_limits<unsigned int>::max());
 
+        for (const unsigned int grain_id : old_grains_per_op[op])
+          for (const auto &old_segment : old_grains.at(grain_id).get_segments())
+            {
               // Flatten all segments of all grains for the given order
               // parameter
-              new_segments_to_grains.push_back(grain_id);
+              old_segments_to_grains.push_back(grain_id);
 
               boxes.emplace_back(
-                create_bounding_box_around_point(new_segment.get_center(),
-                                                 new_segment.get_radius()));
+                create_bounding_box_around_point(old_segment.get_center(),
+                                                 old_segment.get_radius()));
             }
 
         const auto tree = pack_rtree_of_indices(boxes);
 
-        for (const unsigned int grain_id : old_grains_per_op[op])
-          for (const auto &old_segment : old_grains.at(grain_id).get_segments())
+        for (const unsigned int grain_id : new_grains_per_op[op])
+          for (const auto &new_segment : new_grains.at(grain_id).get_segments())
             {
               const auto box =
-                create_bounding_box_around_point(old_segment.get_center(),
-                                                 old_segment.get_radius());
+                create_bounding_box_around_point(new_segment.get_center(),
+                                                 new_segment.get_radius());
 
               std::vector<typename ContainerType::size_type> result;
 
               tree.query(bgi::intersects(box) &&
-                           bgi::nearest(old_segment.get_center(), 1),
+                           bgi::nearest(new_segment.get_center(), 1),
                          std::back_inserter(result));
 
               // If any of the grain was identified, then there is no need to
               // check for the others, so we break the iteration
               if (!result.empty())
                 {
-                  new_grains_to_old.at(new_segments_to_grains[result[0]]) =
-                    grain_id;
+                  const unsigned int old_grain_id =
+                    old_segments_to_grains[result[0]];
+                  new_grains_to_old.at(grain_id) = old_grain_id;
```

Comment:
Could you describe how the new algorithm is better than the old one?

Reply:
I have updated the description of the issue accordingly.